### PR TITLE
Update BigQueryIO Documentation

### LINKED ISSUE #27298: [Task]: Update the document for BigQueryIO features
### What needs to happen?

**Auto Sharding**

Current document mentioned the autosharding with Storage Write API is not supported. 
image
https://beam.apache.org/documentation/io/built-in/google-bigquery/#exactly-once-semantics

But this PR make AutoSharding available, and `Experimental` annotation was removed in Apache Beam 2.48.0.
https://github.com/apache/beam/pull/16795

I think it's time to announce that BigQueryIO can use Storage Write API with `withAutoSharding`!

**ErrorHandling**

Current documentation mentioned `WriteResult.getFailedInserts` does not work for Storage Write API. 

image
https://beam.apache.org/documentation/io/built-in/google-bigquery/#exactly-once-semantics

But now WriteResult has another function `#getFailedStorageApiInserts`. I think it is better for users to mention this function to rescue failed rows.

https://github.com/apache/beam/blob/v2.48.0/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java#L214

---

I'll create a PR if I can do that, please let me know. Thank you in advance.👋

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by RyuSA**
- Updated the description regarding failed rows for Storage Write API.
- Made `PCollection` formatting consistent.

**Please** add a meaningful description for your change here

fixes #27298 

Current [documentation](https://beam.apache.org/documentation/io/built-in/google-bigquery/#exactly-once-semantics) mentions `WriteResult.getFailedInserts` does not work for Storage Write API. But now `WriteResult.getFailedStorageApiInserts` will return the failed rows for Storage Write API. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by RyuSA** [comment]
assign to next reviewer

**Comment by RyuSA** [comment]
@kennknowles Could you take a look at this PR?

**Comment by liferoad** [comment]
R: @ahmedabu98

File: website/www/site/content/en/documentation/io/built-in/google-bigquery.md [hunk 1]
```diff
@@ -883,9 +883,9 @@ explicitly enable this using [`withAutoSharding`](https://beam.apache.org/releas
 ***Note:*** Auto sharding with `STORAGE_WRITE_API` is supported on Dataflow's legacy runner, but **not** on Runner V2
 {{< /paragraph >}}
 
-When using `STORAGE_WRITE_API`, the PCollection returned by
-[`WriteResult.getFailedInserts`](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.html#getFailedInserts--)
-will not contain the failed rows. If there are data validation errors, the
+When using `STORAGE_WRITE_API`, the `PCollection` returned by
+[`WriteResult.getFailedStorageApiInserts`](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.html#getFailedStorageApiInserts--)
+will contain the rows that failed to be written by the Storage Write API. If there are data validation errors, the
```
Comment (ahmedabu98) [inline website/www/site/content/en/documentation/io/built-in/google-bigquery.md:888]:
```suggestion
will contain the rows that failed to be written to the Storage Write API sink. If there are data validation errors, the
```

**Review by ahmedabu98 — CHANGES_REQUESTED** [review]
Couple nits

Also I think we can remove the sentence `If there are data validation errors, the transform will throw a RuntimeException.` That doesn't happen anymore, data validation errors should also go to the failed inserts DLQ

**Comment by RyuSA** [comment]
@ahmedabu98 
Thank you. 👏 

> Also I think we can remove the sentence `If there are data validation errors, the transform will throw a RuntimeException.` That doesn't happen anymore, data validation errors should also go to the failed inserts DLQ

QQ I thought the "data validation errors" is this shema validation, isn't it? If no I'll quick fix the document.👍

https://github.com/apache/beam/blob/v2.50.0/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiLoads.java#L112

**Comment by ahmedabu98** [comment]
> QQ I thought the "data validation errors" is this schema validation, isn't it? 
I think data validation refers to schema validation, but I don't think that specific link is related. R rows with mismatched schema do go to the DLQ. In practice most failed rows in the DLQ are due to this. I'm not sure how the behavior was before, but we don't fail the transform/pipeline now.

I think the line in the link you sent checks that the coder for the `StorageApiWritePayload` object exists (this does not validate data schema).

**Review by ahmedabu98 — APPROVED** [review]
LGTM (should probably remove that sentence tho)

**Comment by RyuSA** [comment]
@ahmedabu98 Sorry for the delay in getting back to you.

> (should probably remove that sentence

Yes, your comment resolved my QQ . The sentences should be removed. Thanks for letting me know.👍

I have added a new commit.👀.

**Pull request merged**